wrangler-core/src/main/java/io/cdap/wrangler/utils/ArithmeticOperations.java

```diff
@@ -24,380 +24,462 @@
-      return null;
+  private static NumberType getOutputType(Number... nums) throws DirectiveExecutionException {
+    NumberType outputType = NumberType.SHORT;
+    for (Number num : nums) {
+      int outputTypePriority = java.util.Arrays.asList(NumberType.values()).indexOf(outputType);
+      if (num instanceof Byte) {
+        throw new DirectiveExecutionException("Input cannot be of type 'Byte'.");
+      } else if (num == null) {
+        return null;
+      } else if (num instanceof Integer && outputTypePriority == 0) {
+        outputType = NumberType.INTEGER;
+      } else if (num instanceof Long && outputTypePriority <= 1) {
+        outputType = NumberType.LONG;
+      } else if (num instanceof Float && outputTypePriority <= 2) {
+        outputType = NumberType.FLOAT;
+      } else if (num instanceof Double && outputTypePriority <= 3) {
+        outputType = NumberType.DOUBLE;
+      } else if (num instanceof BigDecimal) {
+        outputType = NumberType.BIGDECIMAL;
+      }
     }
-    return x / y;
+    return outputType;
   }
 
   /**
-   * Arithmetic operation - Divide a column by another and return the quotient.
+   * Arithmetic operation - Find the sum of any number of columns, of any valid types.
+   * Output type is most general of input types.
+   * Returns null if any input value is null.
    */
-  public static BigDecimal divideq(BigDecimal x, BigDecimal y) {
-    if (x == null || y == null || y.equals(BigDecimal.valueOf(0))) {
+  public static Number add(Number... nums) throws DirectiveExecutionException {
+    NumberType outputType = getOutputType(nums);
+    if (outputType == null) {
       return null;
     }
-    return x.divide(y, BigDecimal.ROUND_HALF_EVEN).stripTrailingZeros();
-  }
-
-  /**
-   * Arithmetic operation - Divide a column by another and return the remainder.
-   */
-  public static Integer divider(Integer x, Integer y) {
-    if (x == null || y == null || y == 0) {
-      return null;
+    switch (outputType) {
+      case SHORT:
+        short shortSum = 0;
+        for (Number num : nums) {
+          shortSum += num.shortValue();
+        }
+        return shortSum;
+      case INTEGER:
+        int intSum = 0;
+        for (Number num : nums) {
+          intSum += num.intValue();
+        }
+        return intSum;
+      case LONG:
+        long longSum = 0;
+        for (Number num : nums) {
+          longSum += num.longValue();
+        }
+        return longSum;
+      case FLOAT:
+        float floatSum = 0;
+        for (Number num : nums) {
+          floatSum += num.floatValue();
+        }
+        return floatSum;
+      case DOUBLE:
+        double doubleSum = 0;
+        for (Number num : nums) {
+          doubleSum += num.doubleValue();
+        }
+        return doubleSum;
+      default:
+        BigDecimal bdSum = BigDecimal.valueOf(0);
+        for (Number num : nums) {
+          bdSum = bdSum.add(num instanceof BigDecimal ? (BigDecimal) num : new BigDecimal(num.toString()));
```

Comment:
this function is repeated a lot. Make this a `numberToBigDecimal(Number number) function that performs this internally. that way the code is cleaner and more maintainable.

Reply:
Good idea -- fixed!